exporter.py

```diff
@@ -0,0 +1,203 @@
+
+        print("Jira Worklogs")
+        print("=============")
+        print("Create")
+        report_log(to_create)
+
+        print()
+        print("Delete")
+        report_log(to_delete)
+
+        print()
+        print("Odoo Attendances")
+        print("================")
+        for day, day_attendances \
+                in groupby(attendances, key=lambda e: e[0].date()):
+            print("{}".format(day))
+            for attendance in day_attendances:
+                print("  {} → {}".format(
+                    attendance[0].time(),
+                    attendance[1] and attendance[1].time()
+                ))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="gtimelog_exporter options")
+
+    parser.add_argument('-c', '--config',
+                        default=DEFAULT_CONFIG_PATH, type=str)
+    parser.add_argument('-w', '--week',
+                        default=Utils.current_weeknumber(), type=int)
+    parser.add_argument('-y', '--year',
+                        default=Utils.current_year(), type=int)
+    parser.add_argument('--no-interactive', action='store_true')
+
+    args = parser.parse_args()
+
+    config = Utils.parse_config(args)
+
+    odoo_password = tempo_password = env.get('ALL_PASSWORD')
+
+    odoo_password = odoo_password or env.get('ODOO_PASSWORD')
+    if not odoo_password:
+        if args.no_interactive:
+            raise Exception('Password missing in non-interactive, '
+                            'set with ODOO_PASSWORD')
+        odoo_password = getpass('Odoo password: ')
+
+    tempo_password = tempo_password or env.get('TEMPO_PASSWORD')
+    if not tempo_password:
+        if args.no_interactive:
+            raise Exception('Password missing in non-interactive, '
+                            'set with TEMPO_PASSWORD')
+        tempo_password = getpass('Tempo (Jira) password: ')
+
+    config['odoo_password'] = odoo_password
+    config['tempo_password'] = tempo_password
+
+    not_before = datetime.strptime('2019-04-01', '%Y-%M-%d').date()
+    if config['date_window'].start.date() < not_before:
+        raise Exception('This script is not intended to manage attendences '
+                        'prior to April 1st, 2019')
+
+    tempo = TempoClient(config)
+    tempo_logs = tempo.get_worklogs(config['date_window'])
+
+    gt_parser = GtimelogParser(config)
+    attendances, gt_logs = gt_parser.get_entries(config['date_window'])
+
+    to_create = []
+    to_delete = []
+
+    for l in tempo_logs:
+        if l not in gt_logs:
+            to_delete.append(l)
+
+    for l in gt_logs:
+        if l not in tempo_logs:
+            to_create.append(l)
+
+    Utils.report(to_create, to_delete, attendances)
+
+    if args.no_interactive or Utils.ask_confirmation():
+        for l in to_create:
+            tempo.create_worklog(l)
+
+        for l in to_delete:
+            tempo.delete_worklog(l)
```

Comment:
Why not update the entries when only the duration changes?

Reply:
That's on the roadmap ;)
I didn't have enough time to enhance it further